Source/CountDown.spoon/init.lua

```diff
@@ -43,45 +133,145 @@ end
 --- Parameters:
 ---  * minutes - How many minutes
 
-local function canvasCleanup()
-    if obj.timer then
-        obj.timer:stop()
-        obj.timer = nil
-    end
+local function canvas_cleanup()
     obj.canvas[1].frame.w = "0%"
     obj.canvas[2].frame.x = "0%"
     obj.canvas[2].frame.w = "0%"
     obj.canvas:frame({x=0, y=0, w=0, h=0})
 end
 
+local function timers_cleanup()
+   if obj.timerBar then
+      obj.timerBar:stop()
+      obj.timerBar = nil
+   end
+   if obj.timer then
+      obj.timer:stop()
+      obj.timer = nil
+   end
+end
+
+function obj:reset()
+   canvas_cleanup()
+   timers_cleanup()
+   
+   obj.menu:setTitle(obj.iconIdle)
+   obj:menu_items_start()
+   obj.timerRunning = false
+   
+   if not obj.alwaysShow then
+      obj.menu:removeFromMenuBar()
+   end
+end
+
+-- update the timer in the status bar
+--
+function obj:updateTimerString()
+   local minutes = math.floor(self.timeLeft / 60)
+   local seconds = self.timeLeft - (minutes * 60)
+   local timerString = string.format("%s %02d:%02d", obj.currentIcon, minutes, seconds)
+   self.menu:setTitle(timerString)
+end
+
+function obj:create_bar_timer(minutes)
+   local mainScreen = hs.screen.primaryScreen()
+   local mainRes = mainScreen:fullFrame()
+   obj.canvas:frame({x=mainRes.x, y=mainRes.h-obj.canvasHeight, w=mainRes.w, obj.canvasHeight})
```

Comment:
is here `h=` missing?

```suggestion
   obj.canvas:frame({x=mainRes.x, y=mainRes.h-obj.canvasHeight, w=mainRes.w, h=obj.canvasHeight})
```